gh-99547: Add isjunction methods for checking if a path is a junction (#99548)
python/cpython: Doc/library/os.rst

      This method can raise :exc:`OSError`, such as :exc:`PermissionError`,
      but :exc:`FileNotFoundError` is caught and not raised.

   .. method:: is_junction()

      Return ``True`` if this entry is a junction (even if broken);

>       return ``False`` if the entry points to a directory or any kind of file,
>       or if it doesn't exist anymore.

Reviewer: ```suggestion
      return ``False`` if the entry points to a regular directory or file, a
      symlink, or if it doesn't exist anymore.
```

Unlike directory symlinks, junctions are reported as directories. They're mount points, after all.

Note however that junctions can be deleted like files via `os.unlink()`, for which we call WinAPI `RemoveDirectoryW()`. I think that's POSIX-ly correct because the spec for `unlink()` says "[t]he path argument shall not name a directory unless the process has appropriate privileges and the implementation supports using unlink() on directories". One could argue that an implicit privilege enables `unlink()` to delete junctions.

>       return ``False`` if the entry points to a regular directory or file, a
>       symlink, or if it doesn't exist anymore.

Reviewer: Oops, I dropped the "any kind" distinction for files.

```suggestion
      return ``False`` if the entry points to a regular directory, any kind
      of file, a symlink, or if it doesn't exist anymore.
```
